the-name-card/src/components/EditProfile/EditProfile.tsx

```diff
@@ -0,0 +1,309 @@
+import { ChangeEvent, useRef, useState } from 'react';
+import { ChromePicker, ColorResult } from 'react-color';
+import OutsideClickHandler from 'react-outside-click-handler';
+import { IoMdArrowBack } from 'react-icons/io';
+import { FaRegAddressCard } from 'react-icons/fa';
+import {
+	IoColorPaletteOutline,
+	IoClose,
+	IoPersonOutline,
+} from 'react-icons/io5';
+import { LuImagePlus } from 'react-icons/lu';
+import { BiMessageDetail } from 'react-icons/bi';
+import {
+	MdOutlineWorkOutline,
+	MdOutlineEmail,
+	MdPhone,
+	MdDragIndicator,
+} from 'react-icons/md';
+import SortableList, { SortableItem, SortableKnob } from 'react-easy-sort';
+import arrayMove from 'array-move';
+
+import {
+	Dropdown,
+	DropdownItem,
+	WidgetInitialValue,
+	WidgetItem,
+	WidgetProp,
+	WidgetType,
+} from '..';
+import './EditProfile.scss';
+
+export const EditProfile = () => {
+	// banner stuff for future refactor
+	const [bannerColor, setBannerColor] = useState<string>('#10A5F5');
+	const [showColorPalette, setShowColorPalette] = useState(false);
+	const [bannerImageUri, setBannerImageUri] = useState<string | null>(null);
+	const [widgetProperties, setWidgetProperties] = useState<WidgetProp[]>([]);
+
+	const bannerPictureInputRef = useRef<HTMLInputElement>(null);
+
+	const onColorChange = (color: ColorResult) => {
+		setBannerColor(color.hex);
+		setBannerImageUri(null);
+	};
+
+	const toggleColorPalette = () => {
+		setShowColorPalette((prev) => !prev);
+	};
+
+	const openBannerFileInput = () => {
+		bannerPictureInputRef.current?.click();
+	};
+
+	const handleBannerFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+		if (event.currentTarget.files === null) return;
+		const files = event?.target?.files;
+		if (files) {
+			const file = files[0];
+			const url = URL.createObjectURL(file);
+			setBannerImageUri(url);
+		}
+	};
+
+	const bannerStyle = { backgroundColor: bannerColor };
+
+	// profile picture for future refactor
+	const [profilePictureUri, setProfilePictureUri] = useState<string | null>(
+		null
+	);
+	const profilePictureInputRef = useRef<HTMLInputElement>(null);
+
+	const handleProfileFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+		if (event.currentTarget.files === null) return;
+		const files = event?.target?.files;
+		if (files) {
+			const file = files[0];
+			const url = URL.createObjectURL(file);
+			setProfilePictureUri(url);
+		}
+	};
+
+	const openProfileFileInput = () => {
+		profilePictureInputRef.current?.click();
+	};
+
+	const onSortEnd = (oldIndex: number, newIndex: number) => {
+		setWidgetProperties((array) => arrayMove(array, oldIndex, newIndex));
+	};
+
+	const setWidgetPropertiesWrapper = (
```

Comment:
personally, since this function is returning a function, i would name it smth else to make that clear
e.g. `generateWidgetPropertiesSetter` or `generateSetWidgetProperties`)

Reply:
sounds good, I rarely work with higher-order functions so didn't know about the naming convention